src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3451,6 +3452,9 @@ int sockinfo_tcp::setsockopt(int __level, int __optname,
 	bool supported = true;
 	bool allow_privileged_sock_opt = false;
 
+	if (m_sock_state <= TCP_SOCK_ACCEPT_READY && __optval != NULL && is_inherited_option(__level, __optname))
+		m_socket_options_list.push_back(new socket_option_t(__level, __optname,__optval, __optlen));
+
```

Comment:
I dont agree with this change.
Lets say we are passing invalid argument. so each establish socket will reset this option during creation.